### PR TITLE
fix: pass file content-type

### PR DESCRIPTION
**Pull request by romGilad59**

### CODE REVIEW

File: ai21/clients/studio/resources/studio_library.py [hunk 1]
```diff
@@ -28,7 +30,13 @@ def create(
         **kwargs,
     ) -> str:
         url = f"{self._client.get_base_url()}/{self._module_name}"
-        files = {"file": open(file_path, "rb")}
+
+        file_name = os.path.basename(file_path)
+        content_type, _ = mimetypes.guess_type(file_path)
+        if content_type is None:
+            raise ValueError(f"Failed to derive content type for file: {file_name}")
```
Comment (asafgardin) [inline ai21/clients/studio/resources/studio_library.py:37]: Looks good.

A few questions - 
1. Why not also remove the `elif` condition we talked about in the client? Do we still need it?
2. Im pretty sure the `mimetypes` package "guesses" the mime type according to the file extension so if you pass a .zip that is actually a .json, you will get the wrong content-type. (not sure if that's something to consider) Are you familiar with this library? https://github.com/ahupp/python-magic I think it gets the file type from the headers
3. Please extract this logic to its own function
4. Add tests under `/tests/unittests/resources/test_studio_library.py` (this file test does not exist) Im not expecting testing the entire package of course, just a use case where content-type is None and one that isn't :)

Comment (Asaf-m) [reply ai21/clients/studio/resources/studio_library.py:37]: Regarding python-magic - I'm not sure it's a good fit for SDK. Using it means you require implicitly the installation of `libmagic`.

My 2 cents :)

Comment (asafgardin) [reply ai21/clients/studio/resources/studio_library.py:37]: @Asaf-m I agree. Perhaps there are other libraries that get yield similar results?

Comment (asafgardin) [reply ai21/clients/studio/resources/studio_library.py:37]: @romGilad59 Why do we need to explicitly pass the content-type? doesn't the server interprets it automatically when the request arrives as part of the request headers?

Comment (Asaf-m) [reply ai21/clients/studio/resources/studio_library.py:37]: I'm not familiar with other libraries.
As an alternative we can validate the file type on the back end. We're not dealing with malicious clients here (and in fact a malicious client can easily trick python-magic as well), only files that a have wrong suffix.

**Comment by asafgardin** [comment]
@romGilad59 Closing this PR as it is not relevant for now.
When we switch and use `httpx` as oppose from `requests` we would get this logic out of the box.

**Pull request closed without merging**